src/engine-actions/get-payment-preimage.js

```diff
@@ -27,62 +20,82 @@ const POLLING_INTERVAL = 10000
  * then return the hash.
  *
  * @param  {string} paymentHash - Base64 encoded payment hash
- * @returns {GetPaymentPreimageOutcome}
+ * @returns {Promise<GetPaymentPreimageOutcome>}
  * @throws {Error} If payment is in an unknown status
  */
 async function getPaymentPreimage (paymentHash) {
-  const { client, logger } = this
-  const { status } = await lookupPaymentStatus(paymentHash, { client })
+  return new Promise(async (resolve, reject) => {
+    try {
+      if (!paymentHash) {
+        throw new Error('paymentHash must be defined')
+      }
 
-  if (status === PAYMENT_STATUSES.GROUNDED) {
-    return { permanentError: `No payment with hash '${paymentHash}' is in-flight or complete` }
-  }
+      // TODO: look into how gRPC wants `bytes` in JS
```

Comment:
our [settings](https://github.com/sparkswap/lnd-engine/blob/master/src/utils/load-proto.js#L20) have `bytes` as Base64 strings.

Reply:
Perfect, thank you.